File: passwordchecker.py
import re

def check_password_strength(password):
    score = 0
    feedback = []

    if len(password) >= 8:
        score += 1
    else:
        feedback.append("Password should be at least 8 characters long.")

    if re.search(r"[A-Z]", password):
        score += 1
    else:
        feedback.append("Include at least one uppercase letter.")

    if re.search(r"[a-z]", password):
        score += 1
    else:
        feedback.append("Include at least one lowercase letter.")

    if re.search(r"[0-9]", password):
        score += 1
    else:
        feedback.append("Include at least one number.")

    if re.search(r"[^A-Za-z0-9]", password):
        score += 1
    else:
        feedback.append("Include at least one special character.")

    strength_levels = ["Very Weak", "Weak", "Moderate", "Strong", "Very Strong"]
    print(f"\nPassword Strength: {strength_levels[score - 1] if score > 0 else 'Too short'}")

    if feedback:
        print("Suggestions:")
        for f in feedback:
            print(f"- {f}")

if __name__ == "__main__":
    user_input = input("Enter a password to check its strength: ")
    check_password_strength(user_input)

File: passwordchecker.js
<!DOCTYPE html>
<html>
<head>
  <title>Password Strength Checker</title>
  <style>
    body { font-family: Arial; padding: 20px; background: #1e004d; color: #fff; }
    input { padding: 10px; width: 300px; }
    .strength { font-weight: bold; margin-top: 10px; }
  </style>
</head>
<body>

  <h2>Password Complexity Checker</h2>
  <input type="password" id="password" placeholder="Enter password" onkeyup="checkStrength()" />
  <div class="strength" id="strengthMessage"></div>

  <script>
    function checkStrength() {
      const password = document.getElementById("password").value;
      const message = document.getElementById("strengthMessage");

      let strength = 0;
      if (password.length >= 8) strength++;
      if (/[A-Z]/.test(password)) strength++;
      if (/[a-z]/.test(password)) strength++;
      if (/[0-9]/.test(password)) strength++;
      if (/[^A-Za-z0-9]/.test(password)) strength++;

      const strengths = ["Very Weak", "Weak", "Moderate", "Strong", "Very Strong"];
      message.textContent = `Strength: ${strengths[strength - 1] || "Too short"}`;
      message.style.color = ["red", "orange", "gold", "lime", "lightgreen"][strength - 1] || "gray";
    }
  </script>

</body>
</html>
